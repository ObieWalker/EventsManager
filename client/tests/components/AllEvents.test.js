import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
// import configureStore from 'redux-mock-store';
import { AllEvents }
  from '../../src/components/AdminDashboard/AllEvents.jsx';
import { events } from '../__mocks__/centersData';

let mountedComponent;
let moreEvents;
const allEvents = {
  events: [events],
  moreEvents: true
};
let props;

const getAllEvents = jest.fn(() => Promise.resolve({}));
// const getMoreEvents = jest.fn();
// const location = {};
const cancelEvent = jest.fn();

// const mockStore = configureStore();
let wrapper;

const getComponent = () => {
  if (!mountedComponent) {
    props = {
      allEvents,
      getAllEvents,
      cancelEvent,
      moreEvents,
      events
    };
    mountedComponent = shallow(<AllEvents {...props} />);
  }
  return mountedComponent;
};

describe('Admin Events Component', () => {
  beforeEach(() => {});
  wrapper = getComponent();

  it('component successfully rendered', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('testing component will recieve props', () => {
    const nextProps = {
      allEvents: {
        fetchedEvents: [events]
      }
    };

    wrapper = getComponent();
    wrapper.instance().componentWillReceiveProps(nextProps);
    expect(wrapper.state().isLoading).toEqual(false);
  });

  it('testing load more function', () => {
    wrapper = getComponent();
    wrapper.instance().loadMoreContent();
    expect(wrapper.state().pageNo).toEqual(2);
    expect(wrapper.state().limit).toEqual(6);
  });


//   describe('Connected Component', () => {
//     describe('Connected Events Component', () => {
//       it('component successfully rendered', () => {
//         const store = mockStore({
//           events
//         });
//         wrapper = shallow(<ConnectedCenter
//           store={store}
//           allEvents={allEvents}
//           moreEvents={allEvents.moreEvents}
//         />);
//         expect(wrapper.length).toBe(1);
//       });
//     });
//   });
});
