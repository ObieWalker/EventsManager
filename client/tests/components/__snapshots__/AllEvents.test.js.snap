// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Events Component component successfully rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AllEvents
    allEvents={
      Object {
        "events": Array [
          undefined,
        ],
        "moreEvents": true,
      }
    }
    cancelEvent={[MockFunction]}
    getAllEvents={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
            6,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Promise {},
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          All Events and details.
        </h3>,
        <div>
          There are no events booked.
          <ScrollUp
            duration={250}
            easing="easeOutCubic"
            showUnder={100}
            style={
              Object {
                "bottom": 50,
                "cursor": "pointer",
                "position": "fixed",
                "right": 30,
                "transitionDelay": "0s",
                "transitionDuration": "0.2s",
                "transitionTimingFunction": "linear",
              }
            }
            topPosition={0}
          >
            <button
              className="btn btn-floating btn-rounded waves-effect"
              type="button"
            >
              TOP
            </button>
          </ScrollUp>
          <br />
          <br />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "All Events and details.",
        },
        "ref": null,
        "rendered": "All Events and details.",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "There are no events booked.",
            false,
            <ScrollUp
              duration={250}
              easing="easeOutCubic"
              showUnder={100}
              style={
                Object {
                  "bottom": 50,
                  "cursor": "pointer",
                  "position": "fixed",
                  "right": 30,
                  "transitionDelay": "0s",
                  "transitionDuration": "0.2s",
                  "transitionTimingFunction": "linear",
                }
              }
              topPosition={0}
            >
              <button
                className="btn btn-floating btn-rounded waves-effect"
                type="button"
              >
                TOP
              </button>
            </ScrollUp>,
            undefined,
            <br />,
            <br />,
          ],
        },
        "ref": null,
        "rendered": Array [
          "There are no events booked.",
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="btn btn-floating btn-rounded waves-effect"
                type="button"
              >
                TOP
              </button>,
              "duration": 250,
              "easing": "easeOutCubic",
              "showUnder": 100,
              "style": Object {
                "bottom": 50,
                "cursor": "pointer",
                "position": "fixed",
                "right": 30,
                "transitionDelay": "0s",
                "transitionDuration": "0.2s",
                "transitionTimingFunction": "linear",
              },
              "topPosition": 0,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "TOP",
                "className": "btn btn-floating btn-rounded waves-effect",
                "type": "button",
              },
              "ref": null,
              "rendered": "TOP",
              "type": "button",
            },
            "type": [Function],
          },
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            All Events and details.
          </h3>,
          <div>
            There are no events booked.
            <ScrollUp
              duration={250}
              easing="easeOutCubic"
              showUnder={100}
              style={
                Object {
                  "bottom": 50,
                  "cursor": "pointer",
                  "position": "fixed",
                  "right": 30,
                  "transitionDelay": "0s",
                  "transitionDuration": "0.2s",
                  "transitionTimingFunction": "linear",
                }
              }
              topPosition={0}
            >
              <button
                className="btn btn-floating btn-rounded waves-effect"
                type="button"
              >
                TOP
              </button>
            </ScrollUp>
            <br />
            <br />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "All Events and details.",
          },
          "ref": null,
          "rendered": "All Events and details.",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "There are no events booked.",
              false,
              <ScrollUp
                duration={250}
                easing="easeOutCubic"
                showUnder={100}
                style={
                  Object {
                    "bottom": 50,
                    "cursor": "pointer",
                    "position": "fixed",
                    "right": 30,
                    "transitionDelay": "0s",
                    "transitionDuration": "0.2s",
                    "transitionTimingFunction": "linear",
                  }
                }
                topPosition={0}
              >
                <button
                  className="btn btn-floating btn-rounded waves-effect"
                  type="button"
                >
                  TOP
                </button>
              </ScrollUp>,
              undefined,
              <br />,
              <br />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "There are no events booked.",
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="btn btn-floating btn-rounded waves-effect"
                  type="button"
                >
                  TOP
                </button>,
                "duration": 250,
                "easing": "easeOutCubic",
                "showUnder": 100,
                "style": Object {
                  "bottom": 50,
                  "cursor": "pointer",
                  "position": "fixed",
                  "right": 30,
                  "transitionDelay": "0s",
                  "transitionDuration": "0.2s",
                  "transitionTimingFunction": "linear",
                },
                "topPosition": 0,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "TOP",
                  "className": "btn btn-floating btn-rounded waves-effect",
                  "type": "button",
                },
                "ref": null,
                "rendered": "TOP",
                "type": "button",
              },
              "type": [Function],
            },
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
